Warning: null pointer dereference [-Wnull-dereference] in esp32-hal-uart.c

### Board

Irrelevant

### Device Description

Irrelevant

### Hardware Configuration

Irrelevant

### Version

latest master (checkout manually)

### IDE Name

PlatformIO

### Operating System

Windows 10

### Flash frequency

Irrelevant

### PSRAM enabled

no

### Upload speed

Irrelevant

### Description

When compiling the latest master I get the following warning message:

```
In file included from C:/Users/username/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal.h:80,
                 from C:/Users/username/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-uart.c:15:
C:/Users/username/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-uart.c: In function 'uartBegin':
C:/Users/username/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-uart.c:460:51: warning: null pointer dereference [-Wnull-dereference]
         log_e("UART%d initialization error.", uart->num);
```

Due to the fact that I am using PlatformIO-Arduino 6.7.0 the line numbers are slightly different to the current master. The location in the current master is here:
https://github.com/espressif/arduino-esp32/blob/4db2eddd2231f4fca204b2f08e7739bbbb10858a/cores/esp32/esp32-hal-uart.c#L540-L541

### Sketch

```cpp
.
```


### Debug Message

```plain
.
```


### Other Steps to Reproduce

_No response_

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

--- a/cores/esp32/esp32-hal-uart.c
+++ b/cores/esp32/esp32-hal-uart.c
@@ -536,9 +536,9 @@ uart_t *uartBegin(
     retCode &= uartSetPins(uart_nr, rxPin, txPin, UART_PIN_NO_CHANGE, UART_PIN_NO_CHANGE);
   }
   if (!retCode) {
+    log_e("UART%d initialization error.", uart->num);
     uartEnd(uart_nr);
     uart = NULL;
-    log_e("UART%d initialization error.", uart->num);
   } else {
     uartFlush(uart);
     log_v("UART%d initialization done.", uart->num);

